eclair-core/src/test/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3Spec.scala

```diff
@@ -199,4 +203,48 @@ class ChannelCodecs3Spec extends AnyFunSuite {
     }
   }
 
+  test("backward compatibility for htlc status (local)") {
+    val legacyCodec: Codec[Option[Transactions.HtlcTx]] = optional(bool8, htlcTxCodec)
+
+    val inputInfo = InputInfo(OutPoint(randomBytes32(), 42), TxOut(123456L sat, randomBytes(90)), randomBytes(128))
+    val tx = Transaction.read("020000000001018154ecccf11a5fb56c39654c4deb4d2296f83c69268280b94d021370c94e219701000000000000000001d0070000000000002200204adb4e2f00643db396dd120d4e7dc17625f5f2c11a40d857accc862d6b7dd80e0500483045022100d5275b3619953cb0c3b5aa577f04bc512380e60fa551762ce3d7a1bb7401cff9022037237ab0dac3fe100cde094e82e2bed9ba0ed1bb40154b48e56aa70f259e608b01483045022100c89172099507ff50f4c925e6c5150e871fb6e83dd73ff9fbb72f6ce829a9633f02203a63821d9162e99f9be712a68f9e589483994feae2661e4546cd5b6cec007be501008576a91414011f7254d96b819c76986c277d115efce6f7b58763ac67210394854aa6eab5b2a8122cc726e9dded053a2184d88256816826d6231c068d4a5b7c820120876475527c21030d417a46946384f88d5f3337267c5e579765875dc4daca813e21734b140639e752ae67a914b43e1b38138a41b37f7cd9a1d274bc63e3a9b5d188ac6868f6010000")
+
+    val htlcTimeoutTx = HtlcTimeoutTx(inputInfo, tx, 74846, BlockHeight(800000))
+    val htlcSuccessTx = HtlcSuccessTx(inputInfo, tx, randomBytes32(), 4861, BlockHeight(800000))
+
+    val htlc_opts = Map(
+      Option.empty[HtlcTimeoutTx] -> channel.LocalCommitPublished.HtlcOutputStatus.Unknown,
+      Option.empty[HtlcSuccessTx] -> channel.LocalCommitPublished.HtlcOutputStatus.Unknown,
```

Comment:
You're trying to be a bit too fancy with types here: unfortunately both those keys map to the same value, so you're testing only one of these two cases! You should use a `Seq` instead of a `Map` (same below and in `ChannelCodecs2Spec.scala`).

Reply:
WDYM? values don't have to be unique in a `Map`? Indeed both `empty` options will be encoded the same way, but it's part of the test?

Reply:
No, I mean your loop below only tries 3 entries instead of 4, because `Option.empty[HtlcSuccessTx]` replaces `Option.empty[HtlcTimeoutTx]` in your map

Reply:
Done with 7796df559.